Doppler.ContactPolicies.Business.Logic/Services/ContactPoliciesService.cs

```diff
@@ -21,5 +22,36 @@ public async Task<ContactPoliciesSettingsDto> GetContactPoliciesSettingsAsync(st
                 (await _contactPoliciesSettingsRepository.GetContactPoliciesSettingsAsync(accountName)).ToDto();
             return contactPoliciesSettings;
         }
+
+        public async Task<bool> InsertContactPoliciesSettingsAsync(string accountName, ContactPoliciesSettingsDto contactPoliciesSettings)
+        {
+            var existingUsercontactPolicies = await _contactPoliciesSettingsRepository.GetBasicContactPoliciesSettingsAsync(accountName);
+
+            /// User doesnt exist
+            if (existingUsercontactPolicies is null)
+                return false;
+
+            /// User doesnt have permissions
+            if (existingUsercontactPolicies.IdUser is null)
+                throw new Exception($"This action is not allowed for the user with Account {accountName}.");
+
+            var contactPoliciesToInsert = CreateUserContactPoliciesToInsert(contactPoliciesSettings, existingUsercontactPolicies);
```

Comment:
Contact policies update is an idempotent operation, right? 

So, why do you need the old value (`existingUsercontactPolicies`) to generate the new one?

Reply:
Oh, I understand, let's continue the discussion in the other thread: https://github.com/FromDoppler/doppler-contact-policies-api/pull/21/files#r659732883

Marking this as resolved.